src/sentry/web/forms/accounts.py

```diff
@@ -287,9 +267,16 @@ def __init__(self, user, *args, **kwargs):
         if self.user.is_managed:
             # username and password always managed, email and
             # name optionally managed
+            needs_password = True
             for field in ('email', 'name', 'username'):
                 if field == 'username' or field in settings.SENTRY_MANAGED_USER_FIELDS:
                     self.fields[field] = ReadOnlyTextField(label=self.fields[field].label)
+                if field == 'email':
+                    needs_password = False
```

Comment:
actually now that it's outdented i wonder why needs_password is not true by default.